tests/core/factory/test_checkpoint_factory.py

```diff
@@ -415,3 +416,316 @@ def _test_checkpoint_factory_delete_emits_event(self, context):
+
+    @pytest.mark.filesystem
+    def test_add_empty_new_checkpoint__filesystem(self, empty_data_context):
+        return self._test_add_empty_new_checkpoint(empty_data_context)
+
+    @pytest.mark.cloud
+    def test_add_empty_new_checkpoint__cloud(self, empty_cloud_context_fluent):
+        return self._test_add_empty_new_checkpoint(empty_cloud_context_fluent)
+
+    @pytest.mark.unit
+    def test_add_empty_new_checkpoint__ephemeral(self, ephemeral_context_with_defaults):
+        return self._test_add_empty_new_checkpoint(ephemeral_context_with_defaults)
+
+    def _test_add_empty_new_checkpoint(self, context: AbstractDataContext):
+        # arrange
+        checkpoint = Checkpoint(name=self.CHECKPOINT_NAME, validation_definitions=[])
+
+        # act
+        created_checkpoint = context.checkpoints.add_or_update(checkpoint=checkpoint)
+
+        # assert
+        assert created_checkpoint.id
+        context.checkpoints.get(self.CHECKPOINT_NAME)
+
+    @pytest.mark.filesystem
+    def test_add_new_checkpoint_with_validations__filesystem(self, empty_data_context):
+        return self._test_add_new_checkpoint_with_validations(empty_data_context)
+
+    @pytest.mark.cloud
+    def test_add_new_checkpoint_with_validations__cloud(self, empty_cloud_context_fluent):
+        return self._test_add_new_checkpoint_with_validations(empty_cloud_context_fluent)
+
+    @pytest.mark.unit
+    def test_add_new_checkpoint_with_validations__ephemeral(self, ephemeral_context_with_defaults):
+        return self._test_add_new_checkpoint_with_validations(ephemeral_context_with_defaults)
+
+    def _test_add_new_checkpoint_with_validations(self, context: AbstractDataContext):
+        # arrange
+        batch_def = (
+            context.data_sources.add_pandas("data source A")
+            .add_dataframe_asset("asset A")
+            .add_batch_definition_whole_dataframe("batch def A")
+        )
+
+        validation_definitions = [
+            ValidationDefinition(
+                name="val def A",
+                data=batch_def,
+                suite=ExpectationSuite(name="suite A"),
+            ),
+            ValidationDefinition(
+                name="val def B",
+                data=batch_def,
+                suite=ExpectationSuite(name="suite B"),
+            ),
+        ]
+        checkpoint = Checkpoint(
+            name=self.CHECKPOINT_NAME, validation_definitions=validation_definitions
+        )
+
+        # act
+        created_checkpoint = context.checkpoints.add_or_update(checkpoint=checkpoint)
+
+        # assert
+        assert created_checkpoint.id
+        assert len(created_checkpoint.validation_definitions) == len(validation_definitions)
+        for val_def, created_val_def in zip(
+            validation_definitions, created_checkpoint.validation_definitions
+        ):
+            assert created_val_def.id
+            val_def_dict = val_def.dict()
+            val_def_dict["id"] = ANY
```

Comment:
Is ID the only reason we can't just write `assert validation_definitions == created_checkpoint.validation_definitions`? Might read easier if we just make a new list like `val_defs = [vd.replace("id", ANY) for vd in validation_definitions]` and then do the `==`. Just a thought.

Reply:
haha that's actually what i did first, and pydantic barfed because ANY isn't a valid UUID

Reply:
Yet another strike against pydantic!